tests/test_lattice.py

```diff
@@ -1748,3 +2020,545 @@ def test_layering_on_edge_cases():
     layers_single = get_compatible_layers(single_edge)
     assert layers_single == [[(0, 1)]]
     _validate_layers(single_edge, layers_single)
+
+
+def test_layering_on_disconnected_graph():
+    """
+    Tests that the layering algorithm correctly handles a graph consisting
+    of multiple disconnected components.
+    """
+    disconnected_bonds = [
+        (0, 1),
+        (1, 2),
+        (2, 0),
+        (3, 4),
+        (4, 5),
+        (5, 3),
+    ]
+
+    layers = get_compatible_layers(disconnected_bonds)
+
+    # Assert
+    _validate_layers(disconnected_bonds, layers)
+
+    assert len(layers) == 3, "Two disconnected triangles should be 3-edge-colorable."
+
+    layer_with_01 = next(
+        layer for layer in layers if (0, 1) in layer or (1, 0) in layer
+    )
+    assert (3, 4) in layer_with_01 or (4, 3) in layer_with_01
+
+
+# A map from backend name to the expected tensor type.
+BACKEND_TENSOR_MAP = {
+    "numpy": np.ndarray,
+    "jax": jnp.ndarray if jnp else None,
+    "tensorflow": tf.Tensor if tf else None,
+    "pytorch": torch.Tensor if torch else None,
+}
+
+
+@pytest.fixture(scope="function")
```

Comment:
dont use these fixtures, tc has paramterized fixtures for different backend, please refer to other tests files for the usage